nx/include/switch/services/mii.h

```diff
@@ -0,0 +1,177 @@
+/**
+ * @file mii.h
+ * @brief Mii services (mii:*) IPC wrapper.
+ * @author XorTroll
+ * @copyright libnx Authors
+ */
+#pragma once
+#include "../types.h"
+#include "../sf/service.h"
+
+typedef enum {
+    MiiServiceType_System  = 0,  ///< Initializes mii:e.
+    MiiServiceType_User    = 1,  ///< Initializes mii:u.
+} MiiServiceType;
+
+/// Mii age.
+typedef enum {
+    MiiAge_Young   = 0,  ///< Young
+    MiiAge_Normal  = 1,  ///< Normal
+    MiiAge_Old     = 2,  ///< Old
+    MiiAge_All     = 3,  ///< All of them
+} MiiAge;
+
+/// Mii gender.
+typedef enum {
+    MiiGender_Male    = 0,  ///< Male
+    MiiGender_Female  = 1,  ///< Female
+    MiiGender_All     = 2,  ///< Both of them
+} MiiGender;
+
+/// Mii race.
+typedef enum {
+    MiiRace_Black  = 0,  ///< Black
+    MiiRace_White  = 1,  ///< White
+    MiiRace_Asian  = 2,  ///< Asian
+    MiiRace_All    = 3,  ///< All of them
+} MiiRace;
+
+// Mii source flag.
+typedef enum {
+    MiiSourceFlag_Database  = BIT(0),                                          ///< Miis created by the user
+    MiiSourceFlag_Default   = BIT(1),                                          ///< Default console miis
+    MiiSourceFlag_All       = MiiSourceFlag_Database | MiiSourceFlag_Default,  ///< All of them
+} MiiSourceFlag;
+
+// Mii special key code
+typedef enum {
+    MiiSpecialKeyCode_Normal   = 0,           ///< Normal miis
+    MiiSpecialKeyCode_Special  = 0xA523B78F,  ///< Special miis
+} MiiSpecialKeyCode;
+
+typedef struct {
+    Service s;
+} MiiDatabase;
+
+// Mii create ID.
+typedef struct {
+    Uuid uuid;
+} MiiCreateId;
+
+// Mii data structure.
```

Comment:
Are you *sure* this is correct for both nfp and regular-Miis?(sdknso mentions a `Ver3` in various funcs)

Reply:
Most of the mii service stuff I've implemented is based on Ryujinx's implementation, and they provide a similar definition for CreateId as a 0x10 byte struct, not sure about the differences with the Ver3 format